kinematics_base_test/src/test_kinematics_plugin.cpp

```diff
@@ -483,23 +520,121 @@ TEST(IKFastPlugin, getIKMultipleSolutions)
+  robot_model::RobotModelPtr kinematic_model;
+  const boost::shared_ptr<srdf::Model> &srdf_model = rdf_loader_.getSRDF();
+  const boost::shared_ptr<urdf::ModelInterface> &urdf_model =
+      rdf_loader_.getURDF();
+  kinematic_model.reset(new robot_model::RobotModel(urdf_model, srdf_model));
+  robot_model::JointModelGroup *joint_model_group =
+      kinematic_model->getJointModelGroup(
+          kinematics_test.kinematics_solver_->getGroupName());
 
+  // Test inverse kinematics
+  std::vector<double> seed, fk_values;
+  std::vector<std::vector<double>> solutions;
+  double timeout = 5.0;
+  kinematics::KinematicsQueryOptions options;
+  kinematics::KinematicsResult result;
 
-  ROS_INFO_STREAM("Success Rate: "<<(double)success/kinematics_test.num_ik_multiple_tests_);
-  EXPECT_GT(success , 0.99 * kinematics_test.num_ik_multiple_tests_)<<"A total of "<<num_ik_solutions <<" ik solutions were found out of "
-      <<kinematics_test.num_ik_multiple_tests_<<" tests.";
-  ROS_INFO_STREAM("Elapsed time: "<< (ros::WallTime::now()-start_time).toSec());
-}
+  // getIK
+  moveit_msgs::MoveItErrorCodes error_code;
+  std::vector<double> solution_getIK;
+  solution_getIK.resize(
+      kinematics_test.kinematics_solver_->getJointNames().size(), 0.0);
 
+  std::vector<std::string> fk_names;
+  fk_names.push_back(kinematics_test.kinematics_solver_->getTipFrame());
+  robot_state::RobotState kinematic_state(kinematic_model);
 
+  double min = -3.14;
+  double max = 3.14;
+  double rnd;
 
+  unsigned int success = 0;
+  unsigned int num_ik_solutions = 0;
+  ros::WallTime start_time = ros::WallTime::now();
+  for (unsigned int i = 0; i < kinematics_test.num_nearest_ik_tests_; ++i) {
+    seed.resize(kinematics_test.kinematics_solver_->getJointNames().size(),
+                0.0);
+    fk_values.resize(kinematics_test.kinematics_solver_->getJointNames().size(),
+                     0.0);
+    kinematic_state.setToRandomPositions(joint_model_group);
+    kinematic_state.copyJointGroupPositions(joint_model_group, fk_values);
+    std::vector<geometry_msgs::Pose> poses;
+    poses.resize(1);
+
+    for (std::size_t i = 0; i < seed.size(); i++) {
+      double rnd = (double)rand() / (double)RAND_MAX;
+      seed[i] = min + rnd * (max - min);
+    }
+
+    ASSERT_TRUE(kinematics_test.kinematics_solver_->getPositionFK(
+        fk_names, fk_values, poses));
+
+    // uncomment next line to see the behavior change of KDL plugin
+    //    seed = fk_values;
+
+    // getIK
+    solution_getIK.clear();
+    kinematics_test.kinematics_solver_->getPositionIK(
+        poses[0], seed, solution_getIK, error_code);
+    if (error_code.val == error_code.SUCCESS) {
+      double error_getIK = 0.0;
+      for (std::size_t j = 0; j < seed.size(); ++j) {
+        error_getIK = error_getIK + fabs(solution_getIK[j] - seed[j]);
+      }
+
+      solutions.clear();
+      kinematics_test.kinematics_solver_->getPositionIK(poses, seed, solutions,
```

Comment:
I agree with @jrgnicho, you need to comment this whole test better.  It took me quite a while to verify what you are doing (and I know what test you are trying to run).